src/tests/functional/shared_test_classes/src/single_layer/matrix_nms.cpp

```diff
@@ -140,8 +140,13 @@ void MatrixNmsLayerTest::compare(const std::vector<ov::runtime::Tensor> &expecte
         const auto _dims = actual.get_shape();
         if (_dims.size() == 1 && _dims[0] == numBatches) {
             batchIndex = outputIndex;
-            auto buffer = reinterpret_cast<const int32_t*>(actual.data());
-            std::copy_n(buffer, numBatches, numPerBatch.begin());
+            if (actual.get_element_type() == ov::element::i32) {
+                auto buffer = actual.data<int32_t>();
+                std::copy_n(buffer, numBatches, numPerBatch.begin());
+            } else {
+                auto buffer = actual.data<int64_t>();
+                std::copy_n(buffer, numBatches, numPerBatch.begin());
+            }
```

Comment:
can we use copy as void * with `actual.byte_size()`

Reply:
here we copy i32 or i64 data in i32, so we need to perform conversion